src/test/java/org/sagebionetworks/bridge/models/schedules2/adherence/AdherenceStateTest.java

```
@@ -0,0 +1,224 @@
+        metadata = ImmutableList.of(meta1, meta2);
+        
+        StudyActivityEvent e1 = new StudyActivityEvent.Builder()
+                .withEventId("event1")
+                .withTimestamp(EVENT_TS1)
+                .build();
+        
+        StudyActivityEvent e2 = new StudyActivityEvent.Builder()
+                .withEventId("event2")
+                .withTimestamp(EVENT_TS2)
+                .build();
+        
+        List<StudyActivityEvent> events = ImmutableList.of(e1, e2);
+        
+        rec1 = new AdherenceRecord();
+        rec1.setInstanceGuid("ar1");
+        rec1.setEventTimestamp(EVENT_TS1);
+        
+        rec2 = new AdherenceRecord();
+        rec2.setInstanceGuid("ar2");
+        rec2.setEventTimestamp(EVENT_TS2);
+        
+        List<AdherenceRecord> adherenceRecords = ImmutableList.of(rec1, rec2);
+        
+        state = new AdherenceState.Builder()
+            .withMetadata(metadata)
+            .withEvents(events)
+            .withAdherenceRecords(adherenceRecords)
+            .withNow(NOW)
+            .withShowActive(true)
+            .withClientTimeZone(TEST_CLIENT_TIME_ZONE)
+            .build();
+    }
+    
+    @Test
+    public void testConstruction_nulls() {
+        AdherenceState emptyState = new AdherenceState.Builder().build();
+        assertEquals(emptyState.getMetadata(), ImmutableList.of());
+        assertNull(emptyState.getAdherenceRecordByGuid("event1"));
+        try {
+            emptyState.getEventStreamById("event1");    
+        } catch(IllegalArgumentException e) {
+        }
+        try {
+            emptyState.getEventStreamDayByKey(meta1);    
+        } catch(IllegalArgumentException e) {
+        }
+        try {
+            emptyState.getDaysSinceEventById("event1");    
+        } catch(IllegalArgumentException e) {
+        }
+        try {
+            emptyState.getEventTimestampById("event1");    
+        } catch(IllegalArgumentException e) {
+        }
```

I think it would be worth it to add fail(..) statements in the try blocks. Just to prove that the getters aren't hiding the exceptions and returning nulls.

---

Good catch. This test is wrong (I removed the illegal argument exception checks) and fails caught that. Changed the test to reflect the fact that these methods now return null in most cases.